src/app/chat/views/channel.nim

```diff
@@ -93,13 +93,15 @@ QtObject:
     if (self.chats.chats.len == 0): return
     let selectedChannel = self.getChannel(channelIndex)
     if (selectedChannel == nil): return
-    discard self.status.chat.markAllChannelMessagesRead(selectedChannel.id)
+    if (selectedChannel.unviewedMessagesCount > 0 or selectedChannel.mentionsCount > 0):
+      discard self.status.chat.markAllChannelMessagesRead(selectedChannel.id)
```

Comment:
It might be a good idea to try to run this async, or pass exactly the message ids that are not read (there's a function for marking as seen an array of message ids in status-go), since `markAllChannelMessagesRead` is really slow